### DISCUSSION TITLE
Bluetooth mouse and keyboard support for PS2

### DISCUSSION THREAD

**Discussion by YgorSan** [darthcloud/BlueRetro#730] [Ideas]
would it be possible to implement support for other bluetooth peripherals for the PS2 like mouse and keyboard? I saw that there are some consoles that support this like Dreamcast, N64 and PS1 so it could be interesting to have something like that on the PS2, in addition to having many multiplayer games and homebrews that support this, unfortunately as expected they only work via USB cable (or in some cases with some keyboards wireless with its own dongle). Support for bluetooth keyboards and mice would be very viable, especially for those who use online multiplayer game chats on the PS2 and also want to play games like Half-Life but have to worry about incompatibility with recent mouse models. Also, the console only has 2 USB ports, so for those who run their backups and homebrews through USB drives or use them to power devices like the PS2toHDMI, it might be a setback, as even if you use USB hubs, they tend to to not work well on the console and cause other problems reading USB 1.1 ports

**Comment by YgorSan** [comment]
...a fun idea, but maybe you could adapt the DS4 touch to an emulated mouse? LOL

**Comment by hardlevel** [comment]
I upvote this... I tried to connect a bluetooth keyboard, it connects, but nothing happens, the mapping seens correct, but any key is working, would be nice to adapt keyboard and mouse to use instead controllers

**Comment by arkwise** [comment]
pleeease ad keyboard support for the PS2 and PSX. i am running linux on the ps2 and would like to use a bluetooth keyboard and mouse via the controller ports apposed to the USB ports as i need them for other things.

im currently developing linux on the playstation 1 using the 2.4 uclinux kernel.
